### PR TITLE
Proposing new project team structures table and supporting text

### PR DESCRIPTION
**Pull request by jcohen02**
This PR includes a suggested new "Proejct team structures" table based on some of the discussions we've had around this. Not sure if this is quite the right sort of thing and happy to receive comments / suggestions. This is in a separate file for now but we can merge it across if it seems acceptable and can be moved into the main manuscript.

### CODE REVIEW

File: ProjectTeamStructuresNew.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    </tr>
+    <tr>
+      <td>TEAM</td>
+      <td>N/A</td>
+      <td>Will need to work effectively with their home RSE team, as well as, potentially with researchers they are developing code for.</td>
+      
+      <td>Effective teamwork vital - do they have the skills and knowledge to support team-based software development?</td>
+      <td>Will need to be able to work and collaborate effectively in a team, use required tools and processes, infrastructure, etc.</td>
+      
+      <td>Can offer support with team work. Many examples of courses around working effectively in a team, some vvore </td>
```
[the remark this answers is not in the document]
Comment (jcohen02) [reply ProjectTeamStructuresNew.md]: Ah, yes, I spotted that before I committed it and remember removing it and completing the sentence - not sure how it ended up there!! I'll sort that out now...

[the remark this answers is not in the document]
Comment (jcohen02) [reply ProjectTeamStructuresNew.md]: That's been fixed on my end and all changes were already pushed to the PR branch, it seems this is out of date for some reason. If I go back to the PR in the list and view the "Files changed" tab, I don't see that, I also note it says "Outdated" at the top, hopefully you can see the latest version now?

File: ProjectTeamStructuresNew.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      <td>Greater focus on reusability/sharing but likely not part of project aims.</td>
+      
+      <td>May be looking to develop reusable shareable outputs but likely case-by-case basis.</td>
+      <td>Focus on quality and practices, reusability/packaging driven by project needs and spec.</td>
+      
+      <td>May provide policies on sharing and reuse of software. May be driven by funder requirements/policies.</td>
+    </tr>
+    <tr>
+      <td>MOD</td>
+      <td>Need full awareness of codebase to support extension/maintenance. May not additional experience or support. If project taken on from another individual developer, there may be challenges.</td>
```
Comment (jpthiele) [inline ProjectTeamStructuresNew.md:49]: I think there is a have missing in "May not additional"

**Comment by jpthiele** [comment]
Also I am not sure whether the NEW row covers the full idea of the curiosity competency which should be about more than software tools and practices, right?

**Comment by CaptainSifff** [comment]
pandoc converts it nicely using:
`pandoc --from markdown_strict ProjectTeamStructuresNew.md -o test.html `

**Comment by jcohen02** [comment]
> pandoc converts it nicely using: `pandoc --from markdown_strict ProjectTeamStructuresNew.md -o test.html `

Thanks for checking this @CaptainSifff.

**Comment by CaptainSifff** [comment]
Hi @jcohen02 ,
I pushed some changes already to your branches, have a look at it.
On top of that here are some comments:

- I think I reworded it to use "group" for a bunch of independent RSE/researchers that got thrown together and "team" only for the insitutional RSE Team.
- Organisation contains a lot of "may" shouldn't this include more Should and must, sice you wanted to point out clear gaps?
- and I think we can give examples of what represent the columns.

File: ProjectTeamStructuresNew.md (hunk 2 above)
Comment (jcohen02) [reply ProjectTeamStructuresNew.md]: Thanks @jpthiele, this has been fixed by @CaptainSifff (thanks @CaptainSifff)

**Comment by jcohen02** [comment]
> Hi @jcohen02 , I pushed some changes already to your branches, have a look at it. On top of that here are some comments:
> 
>     * I think I reworded it to use "group" for a bunch of independent RSE/researchers that got thrown together and "team" only for the insitutional RSE Team.
> 
>     * Organisation contains a lot of "may" shouldn't this include more Should and must, sice you wanted to point out clear gaps?
> 
>     * and I think we can give examples of what represent the columns.

Thanks for the comments and updates @CaptainSifff. On looking at this again I realise that there's also a lot of "may"s in the intro text above the table which makes it a little awkward to read. I've fixed this. In terms of the text in the organisation column, I'll look at improving this - the use of "may" so many times is because I think these are all optional things or possibilities of how an organisation could contribute rather than actual direct answers or solutions.

I changed the column headings because as much as I knew what I was trying to represent here, when I re-read them, I didn't feel it was very clear what they're trying to show! Hopefully things are a bit clearer now?

**Comment by CaptainSifff** [comment]
Hi @jcohen02 , 
I looked again, at the table and I wondered, whether you can make sure that the facets of the competencies highlight what aspect people should know in the respective roles? To get back to the original intent of working out how much people need to know in different roles.

**Comment by jpthiele** [comment]
I tried grid_tables for a different project. And it is definitely nice to read in the markdown source,
but I couldn't yet figure out a way to tell pandoc to add horizontal and vertical lines in tex.
So for now this would be a somewhat manual process (or a precarious sed operation).

File: ProjectTeamStructuresNew.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      <td>N/A</td>
+      <td>Will need to work effectively with their home RSE team, as well as, potentially with researchers they are developing code for.</td>
+      <td>Effective teamwork vital - do they have the skills and knowledge to support team-based software development?</td>
+      <td>Will need to be able to work and collaborate effectively in a team, use required tools and processes, infrastructure, etc.</td>
+      <td>Can offer support with team work. Many examples of courses around working effectively in a team.</td>
+    </tr>
+</table>
+
+In the table above, we've looked at how different competencies can be related to and handled by researchers and RSEs working in different environments within an organisation and how the organisations themselves can contribute. We recognise that this is a challenging area to gain a detailed view of and that our content in the table is still a significant generalisation. We talk about the "Research Software Engineer" as a single entity but as the field expands, we expect to see more roles and job titles emerging around the RSE concept, many of which fit under the wider umbrella of Research Technology Professionals (RTPs).
```
Comment (knarrff) [inline ProjectTeamStructuresNew.md:147]: Please split this up into multiple lines, one sentence each.

**Comment by jngrad** [comment]
In pandoc 3.0, it will be possible to style the table with colspans and rowspans using `--from markdown+grid_tables` in the pandoc command line ([user guide section](https://github.com/jgm/pandoc/blob/6673f832ea6bb2498f588b78eb6fb8c099fb408a/MANUAL.txt#L4500-L4511)). This should then work with LaTeX, but then GitHub won't be able to render the table as HTML.

**Comment by CaptainSifff** [comment]
@jcohen02 Since you made the wise decision of having it in a separate file, I don't have to merge your contribution.

**Comment by jcohen02** [comment]
> @jcohen02 Since you made the wise decision of having it in a separate file, I don't have to merge your contribution.

Ok 🙂, is the aim then to use this directly in the separate competencies manuscript? We could merge this into the current main doc in place of the simpler table that's currently there but if that makes things more complex due to the use of HTML, happy to keep this separate.

**Comment by CaptainSifff** [comment]
I would want to have it in the main files with markdown syntax. Newer markdown versions allow to express these tables, somehow....

**Comment by jngrad** [comment]
> I would want to have it in the main files with markdown syntax. Newer markdown versions allow to express these tables, somehow....

It's possible to run pandoc 3.1 in GitHub Actions using the MWE in [pandoc/pandoc-action-example](https://github.com/pandoc/pandoc-action-example). See the integration in my branch [`jc_group-competencies`](https://github.com/jngrad/paper_teaching-learning-RSE/tree/jc_group-competencies). As a proof-of-concept, I added metadata in the new Markdown file to make it self-contained in a different branch, but the result was disappointing (see run [6075650131](https://github.com/jngrad/paper_teaching-learning-RSE/actions/runs/6075650131), click on "PDF files " from the artifacts section to download them).

There are several issues:

* the contents of the first column are missing (in this case "DOCBB"), not sure why
* the Markdown file cannot be rendered as a table by GitHub anymore (see [blob](https://github.com/jngrad/paper_teaching-learning-RSE/blob/jc_group-competencies_test/ProjectTeamStructuresNew.md))
* the Markdown lines get extremely long, and modifying one cell requires manually wrapping the contents of all other cells in the row (see [raw blob](https://raw.githubusercontent.com/jngrad/paper_teaching-learning-RSE/jc_group-competencies_test/ProjectTeamStructuresNew.md))
* it's no longer possible to build the PDF files locally without manually installing pandoc 3

I would recommend rewriting this table in LaTeX directly. It addresses all the drawbacks mentioned above, except for the GitHub rendering.

**Comment by CaptainSifff** [comment]
oh well, I wanted to have something that I can easily convert from one format to the other....
But to be honest, how probable is it that we will end up with a journal that does not accept latex source?

**Comment by jpthiele** [comment]
Good or bad to know. 
I also started on converting the table to grid_tables, 
but spend the last minutes of the train ride today just moving information. 
I had planned to test it tomorrow but reading the comment of @jngrad I don't need to continue with that I guess.

**Comment by jpthiele** [comment]
Update:
I just played around a bit with LaTeX and using the package array allows specifying a column width using m{3.5cm} instead of c
in tabular. Then, the individual text is still readable and we don't need manual linebreaks and such.
Additionally, with the multirow package we can achieve the nice table headings with splits into local and RSE-team.

**Comment by CaptainSifff** [comment]
@jpthiele , @jngrad thanks for making it possible to include this!!

**Pull request merged**